core/templates/dev/head/services/speech-synthesis-chunker.service.spec.ts

```diff
@@ -32,87 +32,212 @@ describe('Speech Synthesis Chunker Service', () => {
+      ' speakable text', () => {
+      const html = (
+        '<oppia-noninteractive-link text-with-value="&quot;">' +
+        '</oppia-noninteractive-link>' +
+        '<li>"Speech"</li>' +
+        '<li>Text</li>'
+      );
+
+      expect(speechSynthesisChunkerService.convertToSpeakableText(html))
+        .toBe('Speech. Text. ');
+    });
+  });
+
+  describe('speak', function() {
+    const MockSpeechSynthesisUtteranceConstructor = (
+      SpeechSynthesisUtterance);
+    const mockSpeechSynthesisUtteran = {
+      speak: () => {},
+      onend: () => {}
+    };
+
+    beforeEach(() => {
+      spyOn(window, 'SpeechSynthesisUtterance').and.returnValue(
+        // @ts-ignore mock doesn't have all property and methods of a native
+        // SpeechSynthesisUtterance
+        mockSpeechSynthesisUtteran);
+    });
+
+    it('should not speak when chunk is too short', () => {
+      const speakSpy = spyOn(window.speechSynthesis, 'speak').and
+        .callFake(function() {
+          mockSpeechSynthesisUtteran.onend();
+        });
+      const speechSynthesisUtterance = (
+        new MockSpeechSynthesisUtteranceConstructor('a'));
+      const callbackSpy = jasmine.createSpy('callback');
+      speechSynthesisChunkerService.speak(
+        speechSynthesisUtterance, callbackSpy);
+
+      expect(callbackSpy).toHaveBeenCalled();
+      expect(speakSpy).not.toHaveBeenCalled();
+    });
+
+    it('should not speak when chunk is a falsy value', () => {
+      const speakSpy = spyOn(window.speechSynthesis, 'speak').and
+        .callFake(function() {
+          mockSpeechSynthesisUtteran.onend();
+        });
+      const speechSynthesisUtterance = (
+        new MockSpeechSynthesisUtteranceConstructor(''));
+      const callbackSpy = jasmine.createSpy('callback');
+      speechSynthesisChunkerService.speak(
+        speechSynthesisUtterance, callbackSpy);
+
+      expect(callbackSpy).toHaveBeenCalled();
+      expect(speakSpy).not.toHaveBeenCalled();
+    });
+
+    it('should speak speech twice', () => {
+      // For some reason, just call install doesn't work, we need to uninstall
+      // first so then actually install.
+      jasmine.clock().uninstall();
+      jasmine.clock().install();
+
+      const speakSpy = spyOn(window.speechSynthesis, 'speak').and
+        .callFake(function() {
+          mockSpeechSynthesisUtteran.onend();
+        });
+      const speechSynthesisUtterance = (
+        new MockSpeechSynthesisUtteranceConstructor(
+          'Value inside utterance for testing purposes.' +
+          ' This is the next chunk'));
+      const callbackSpy = jasmine.createSpy('callback');
+      speechSynthesisChunkerService.speak(
+        speechSynthesisUtterance, callbackSpy);
+
+      jasmine.clock().tick(5000);
+
+      expect(callbackSpy).toHaveBeenCalled();
+      expect(speakSpy).toHaveBeenCalledTimes(2);
+
+      jasmine.clock().uninstall();
+    });
+
+    it('should speak once when cancel is requested', () => {
```

Comment:
Change this method as per comment above.

Reply:
Which comment?

Reply:
@DubeySandeep ptal!

Reply:
This [comment](https://github.com/oppia/oppia/pull/8636#discussion_r388668928), (Sorry, I think earlier my comment wasn't published may be because of slow internet connection.)

Reply:
Done.